presto-client/src/main/java/okhttp/internal/tls/DistinguishedNameParser.java

```diff
@@ -0,0 +1,428 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//CHECKSTYLE:OFF
+package okhttp.internal.tls;
+//CHECKSTYLE:ON
+
+import javax.security.auth.x500.X500Principal;
+
+/**
+ * A distinguished name (DN) parser. This parser only supports extracting a string value from a DN.
+ * It doesn't support values in the hex-string style.
+ *
+ * This class is based on the OkHttp library.The original source can be found here:
+ * https://github.com/square/okhttp/blob/okhttp_3.9.x/okhttp/src/main/java/okhttp3/internal/tls/DistinguishedNameParser.java
```

Comment:
This seems to be a branch name, so the file may not be the same down the line. Can you use a permalink to the existing commit on this branch instead?